actors/builtin/miner/miner_actor.go

```diff
@@ -142,70 +140,40 @@ func (a Actor) ChangeWorkerAddress(rt Runtime, params *ChangeWorkerAddressParams
 }
 
 //////////////////
-// SurprisePoSt //
+// FallbackPoSt //
 //////////////////
 
-// Called by Actor to notify StorageMiner of SurprisePoSt Challenge.
-func (a Actor) OnSurprisePoStChallenge(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
-	rt.ValidateImmediateCallerIs(builtin.StoragePowerActorAddr)
-
+// Invoked by miner's worker address to submit their fallback post
+func (a Actor) SubmitFallbackPoSt(rt Runtime, params *abi.OnChainPoStVerifyInfo) *adt.EmptyValue {
 	var st State
-	challenged := rt.State().Transaction(&st, func() interface{} {
-		// If already challenged, do not challenge again.
-		// Failed PoSt will automatically reset the state to not-challenged.
-		if st.PoStState.isChallenged() {
-			return false
-		}
+	rt.State().Transaction(&st, func() interface{} {
+		rt.ValidateImmediateCallerIs(st.Info.Worker)
 
-		st.PoStState = PoStState{
-			LastSuccessfulPoSt:     st.PoStState.LastSuccessfulPoSt,
-			SurpriseChallengeEpoch: rt.CurrEpoch(),
-			NumConsecutiveFailures: st.PoStState.NumConsecutiveFailures,
+		if rt.CurrEpoch() > st.PoStState.ProvingPeriodStart+power.FallbackPostChallengeDuration {
+			rt.Abort(exitcode.ErrIllegalState, "PoSt submission too late")
 		}
-		return true
-	}).(bool)
 
-	if challenged {
-		// Request deferred Cron check for SurprisePoSt challenge expiry.
-		cronPayload := CronEventPayload{
-			EventType: CronEventType_Miner_SurpriseExpiration,
-			Sectors:   nil,
+		if rt.CurrEpoch() <= st.PoStState.ProvingPeriodStart {
+			rt.Abort(exitcode.ErrIllegalState, "Not currently in submission window for PoSt")
 		}
-		surpriseDuration := power.SurprisePostChallengeDuration
-		a.enrollCronEvent(rt, rt.CurrEpoch()+surpriseDuration, &cronPayload)
-	}
-	return &adt.EmptyValue{}
-}
+		a.verifyWindowedPost(rt, &st, params)
```

Comment:
Maybe worth dropping a comment that a failed verification doesn't immediately cause a penalty - the miner has until the end of the window to submit a good proof.